![Y](https://s3.amazonaws.com/dq-content/354/hacker_news.jpg)

#  Exploring Hacker News Posts

## Introduction

**Hacker News** (sometimes abbreviated as HN) is a social news website https://news.ycombinator.com/ focusing on computer science and entrepreneurship. 

It is run by Paul Graham's investment fund and startup incubator, Y Combinator. In general, content that can be submitted is defined as "anything that gratifies one's intellectual curiosity."

The word hacker in "Hacker News" is used in its original meaning and refers to the hacker culture which consists of people who enjoy tinkering with technology.

The intention was to recreate a community similar to the early days of Reddit. 

However, unlike Reddit where new users can immediately both upvote and downvote content, Hacker News does not allow users to downvote content until they have accumulated 501 "karma" points...

https://en.wikipedia.org/wiki/Hacker_News

### Data dictionary:

- `id`: the unique identifier from Hacker News for the post

- `title`: the title of the post

- `url`: the URL that the posts links to, if the post has a URL

- `num_points`: the number of points the post acquired, calculated as the total number of upvotes minus the total number of downvotes

- `num_comments`: the number of comments on the post

- `author`: the username of the person who submitted the post

- `created_at`: the date and time of the post's submission
* * *

# <a id='0'>Index</a>

Users submit **`Ask`** posts to ask the community a specific question, and users also submit **`Show`** posts to show a project, product, or just something interesting.

A number of questions have then been raised to understand a little better the character of the community.


### <a href='#1'>1. Finding posts starting with `Ask` and `Show`</a>

### <a href='#2'>2. Average and maximum number of comments for `Ask` and `Show` posts</a>

### <a href='#3'>3. Titles of the highest score for `Ask` and `Show` posts </a>

### <a href='#4'>4. Finding the number of `Ask` comments posted by hour</a>

### <a href='#5'>5. Posts created at a certain time, receive more comments on average.</a>

### <a href='#6'>6. Most relevant authors in ``Ask` post</a>

### <a href='#7'>7. What are the most significant activity by user in `Ask` post? </a>

### <a href='#8'>8. What are the most significant activity by user in `Show` posts? </a>

### <a href='#9'>9. Who has been the most voted, when, and title of the post?</a>

### <a href='#10'>10. Summary </a>

In [1]:
import matplotlib.pyplot as plt
import datetime as dt

In [2]:
from csv import reader
hn = open('hacker_news.csv')
hn = reader(hn)
hn = list(hn)
header = hn[0]

### Dataset exploration

This is the `header` we are going to work with, corresponds to row 0 on `hn`.

In [3]:
header

['id', 'title', 'url', 'num_points', 'num_comments', 'author', 'created_at']

These are the first 5 rows of our dataset `hn` with which we are going to work, so we can get an idea of the data content and what it looks like.

In [4]:
for filas in hn[1:6]:
    print(filas)

['12224879', 'Interactive Dynamic Video', 'http://www.interactivedynamicvideo.com/', '386', '52', 'ne0phyte', '8/4/2016 11:52']
['10975351', 'How to Use Open Source and Shut the Fuck Up at the Same Time', 'http://hueniverse.com/2016/01/26/how-to-use-open-source-and-shut-the-fuck-up-at-the-same-time/', '39', '10', 'josep2', '1/26/2016 19:30']
['11964716', "Florida DJs May Face Felony for April Fools' Water Joke", 'http://www.thewire.com/entertainment/2013/04/florida-djs-april-fools-water-joke/63798/', '2', '1', 'vezycash', '6/23/2016 22:20']
['11919867', 'Technology ventures: From Idea to Enterprise', 'https://www.amazon.com/Technology-Ventures-Enterprise-Thomas-Byers/dp/0073523429', '3', '1', 'hswarna', '6/17/2016 0:01']
['10301696', 'Note by Note: The Making of Steinway L1037 (2007)', 'http://www.nytimes.com/2007/11/07/movies/07stein.html?_r=0', '8', '2', 'walterbell', '9/30/2015 4:12']


### Data settings

#### Deleting the first row from `hn`.

In [5]:
del hn[0]

<a href='#0'> back to index</a>
* * *
## <a id='1'>1. Finding posts starting with`Ask HN` and `Show HN`</a>

### Extracting post

To find posts starting with  **`Ask HN`** or **`Show HN`**, we will use the string method `startswith`. 


### Filter strategy


Filter the information we will create three empty lists called: 

- **`ask_posts`**

- **`show_posts`** 

- **`other_posts`**.


- Converting titles to lowercase once this process is done, then what will be done is filter by the content we waFilternt with the method `object.startswith('string')`.

We will loop through `hn` and assign the title of each row to a variable called `title` which will be the one that already has the content in lowercase.

- If the lowercase version of the title starts with `ask hn`, add the row to `ask_posts`.

- If the lowercase version of the title starts with `show hn`, add the row to `show_posts`.

- Otherwise, add to `other_post`

In [6]:
header

['id', 'title', 'url', 'num_points', 'num_comments', 'author', 'created_at']

In [7]:
ask_posts = []
show_posts = []
other_posts = []

for fila in hn:
    title = fila[1] #title
    title = title.lower()   # all in lower_case
    
    if title.startswith('ask hn'):
        ask_posts.append(fila)
        
    elif title.startswith('show hn'):
        show_posts.append(fila)
        
    else:
        other_posts.append(fila)
        
l_ask_posts=len(ask_posts)
l_show_posts=len(show_posts)
l_other_posts=len(other_posts)

### Total number of post.

In [8]:
print("`Ask HN` has a number of {ask}, whereas `Show HN` have {show} and `other_posts` {others}".format(ask=l_ask_posts,
                                                                                         show=l_show_posts,
                                                                                             others=l_other_posts))

`Ask HN` has a number of 1744, whereas `Show HN` have 1162 and `other_posts` 17194


<a href='#0'> back to index</a>
* * *
## <a id='2'>2. Average and maximum number of comments for `Ask_HN` and `Show_HN` Posts</a> 

Samples of listing contents `ask_posts` and `show_posts`

The total number of comments on **ask** entries must be assigned to `total_ask_comments`.

**Header looks like:**

    ['id', 'title', 'url', 'num_points', 'num_comments', 'author', 'created_at']
       0      1       2          3             4             5           6

In [9]:
def avg_max_comments(post):
    maxi = 0
    total_ask_comments = 0
    for comment in post:
        total_ask_comments += int(comment[4])  # convert string to int ['num_comments'] and incrementing value
        if maxi <= int(comment[4]): #detector valores mas grandes que maxi (inicialmente 0)
            maxi = int(comment[4])
    
    avg_comments = total_ask_comments / len(post) # That`s the average: Ask comments / Total amount comments 
    print("\n")
    print('avg number of "Ask_posts" comments:', round(avg_comments,2))
    print("\n")
    print('max number of "Ask_posts" comment:', maxi)

### Average and max number of comments for `Ask_HN`

In [10]:
avg_max_comments(ask_posts)



avg number of "Ask_posts" comments: 14.04


max number of "Ask_posts" comment: 947


### Average and max number of comments for `Show_HN`

In [11]:
avg_max_comments(show_posts)



avg number of "Ask_posts" comments: 10.32


max number of "Ask_posts" comment: 306


<a href='#0'> back to index</a>
* * *

## <a id='3'>3. Titles of the highest score for `Ask_HN` and `Show_HN` Posts</a> 

Function that returns the maximum score with the title in descending order.

**Header looks like:**

    ['id', 'title', 'url', 'num_points', 'num_comments', 'author', 'created_at']
       0      1       2          3             4             5           6

In [12]:
def max_comments(post):
    maximum_coments_post = []
    most_important_questions = {}
    maxi = 0 #as a threshold value!

    for comment in post: #comment is the whole row
        if maxi < int(comment[4]): #Only if comment[4] is bigger than maxi, this value will be on dictionary. 
            maxi = int(comment[4]) # update the value 
            # save into dictionary:
            num_comments = int(comment[4])
            most_important_questions[num_comments] = comment[1] # ['title']
            
    most_voted = list(reversed(sorted(most_important_questions.keys()))) # list of most_important_questions 
                                                                     # by keys sorted 

    print("-- Score --            -- Tittle -- ")

    for score in most_voted:
        texto = "    {points}     {relevance}".format(points = score,
                                                      relevance = most_important_questions[score])
        print(texto)
        maximum_coments_post.append(texto)

### Titles of the highest score on  `Ask_HN`.

In [13]:
max_comments(ask_posts)

-- Score --            -- Tittle -- 
    947     Ask HN: Who is hiring? (August 2016)
    910     Ask HN: Who is hiring? (September 2016)
    266     Ask HN: What are the must-read books about economics/finance?
    250     Ask HN: Who wants to be hired? (June 2016)
    234     Ask HN: What are you currently building?
    182     Ask HN: What is your go-to example for a good REST API?
    37     Ask HN: Things you created in 2015?
    33     Ask HN: teaching basic coding and web design offline, solely via iOS devices?
    29     Ask HN: Am I the only one outraged by Twitter shutting down share counts?
    6     Ask HN: How to improve my personal website?


### Titles of the highest score on `Show_HN`.

In [14]:
max_comments(show_posts)

-- Score --            -- Tittle -- 
    306     Show HN: BitKeeper  Enterprise-ready version control, now open-source
    168     Show HN: Nodal. Next-Generation Node.js Server and Framework
    134     Show HN: Download any song without knowing its name
    102     Show HN: Something pointless I made
    22     Show HN: Wio Link  ESP8266 Based Web of Things Hardware Development Platform


<a href='#0'> back to index</a>
* * *
## <a id='4'>4. Finding the number of Ask comments posted by hour</a>.



Ask posts are more likely to receive comments, **we'll focus our analysis just on these posts.**

**We'll determine if `ask posts` created at a certain time are more likely to attract comments**. We'll use the following steps to perform this analysis:

-  <a href='#100 '>Calculate **the number** of `ask posts` in each hour of the day, with the number of comments received.</a>


- <a href='#200 '>Calculate **the average number** of comments ask posts receive by hour created**.</a>

**NOTE**: using the `datetime.strptime()` constructor to **parse dates stored as strings and return datetime objects**, example:

   - `date_1_str = "December 24, 1984"`
   - `date_1_dt = dt.datetime.strptime(date_1_str, "%B %d, %Y")`
   
***Header looks like:***

    ['id', 'title', 'url', 'num_points', 'num_comments', 'author', 'created_at']
       0      1       2          3             4             5           6

Let's use this technique to calculate the number of ask posts created per hour, along with the total number of comments.

In [15]:
result_list = []
num_comment = []
stamp = []
for row in ask_posts:
    time_stamp = row[6]  #created_at
    time_obj = dt.datetime.strptime(time_stamp, "%m/%d/%Y %H:%M")
    stamp.append(time_obj)
    
    comment = int(row[4]) #num_comments
    num_comment.append(comment)
    
    tupla = (time_stamp, num_comment)
    result_list.append(tupla)

In [16]:
result_list[0:5]

[('8/16/2016 9:55',
  [6,
   29,
   1,
   3,
   17,
   1,
   4,
   1,
   1,
   2,
   7,
   1,
   1,
   4,
   4,
   2,
   3,
   1,
   22,
   2,
   2,
   7,
   7,
   3,
   6,
   2,
   1,
   3,
   29,
   2,
   20,
   3,
   3,
   33,
   5,
   4,
   7,
   11,
   1,
   9,
   2,
   37,
   1,
   2,
   4,
   1,
   1,
   182,
   5,
   9,
   8,
   24,
   5,
   3,
   4,
   7,
   10,
   3,
   3,
   4,
   3,
   20,
   1,
   140,
   5,
   3,
   30,
   2,
   2,
   5,
   1,
   17,
   1,
   1,
   5,
   5,
   2,
   8,
   2,
   6,
   22,
   7,
   12,
   1,
   2,
   72,
   2,
   130,
   3,
   2,
   7,
   15,
   2,
   6,
   1,
   1,
   10,
   2,
   15,
   8,
   2,
   1,
   1,
   1,
   6,
   43,
   19,
   5,
   1,
   234,
   5,
   6,
   9,
   1,
   1,
   19,
   3,
   1,
   25,
   7,
   7,
   71,
   3,
   17,
   2,
   5,
   2,
   3,
   61,
   2,
   17,
   2,
   3,
   2,
   1,
   5,
   1,
   13,
   12,
   1,
   1,
   185,
   7,
   2,
   4,
   2,
   2,
   1,
   2,
   2,
   2,
   1,
   22,
   37,
   8,
   1,
   

In [17]:
...plt.hist(stamp)
plt.ticklabel_format(axis='y', style='plain')
plt.show()

SyntaxError: invalid syntax (<ipython-input-17-26d21a1b2d9b>, line 1)

In [ ]:
plt.hist(num_comment)
plt.ticklabel_format(axis='y', style='plain')
plt.show()

Let's work with the above list and create two dictionaries:

- `counts_by_hour`: To know how many times a certain number of hours is repeated.

- `comments_by_hour`: To know the number of times a comment has been created at a given time


Looping through each row of `result_list` tuple, we get the **hour from the date**, the first element of `result_list` using **datetime.strptime()** method to parse the date and create a datetime object and extract the **number of comments** 



Determining whether or not that hour exists in the `counts_by_hour` dictionary.

- If there is no such time we give it the value of 1  and just **add** the value of number of comments to the dictionary `comments_by_hour` with same key value as `counts_by_hour`.


- If this hour exists increase the value by 1 and **increase** the number of comments to the dictionary `comments_by_hour` with same key value as `counts_by_hour`.

In [ ]:
# ('8/28/2016 18:06', '2') <- date and number of comments (tuple)

counts_by_hour   = {}
comments_by_hour = {}
tittle_by_hour = {}

num_points = 0

for row in result_list: # result_list content (-->'8/28/2016 18:06'<--, '2')
    time_stamp = row[0]
    time_obj = dt.datetime.strptime(time_stamp, "%m/%d/%Y %H:%M")
    
    hour = time_obj.hour #Extracting the hour from the `time_stamp`.
    num_comments = row[1]# ('8/28/2016 18:06',-->'2'<--) num coments

    
    if (hour not in counts_by_hour):
        counts_by_hour[hour] = 1 #times per hour
        comments_by_hour[hour] = num_comments # coments por hour

    
    elif hour in counts_by_hour:
        counts_by_hour[hour] += 1 #times per hour
        comments_by_hour[hour] += num_comments # coments por hour

        
print(f"Counts by hour: {counts_by_hour}")
print("---")
print(f"Comments by hour: {comments_by_hour}")

In [ ]:
....%%html
<style>
table {float:left}
</style>

Using these two dictionaries `counts_by_hour` and `comments_by_hour` to calculate the average number of comments for posts created during each hour of the day.

||`counts_by_hour`|
|:---|:---|
|hour|number of times of this hour|
|{9:|45,|
|13:|85,|
|10:|59,|
|...|...|

||`comments_by_hour`|
|:---|:---|
|hour|number of comments in this hour|
|{9:|251,|
|13:|1253,|
|10:|793,|
|...|...|


A list `avg_by_hour` in it we will enter a tuple formed by:

the hour and by the result of average comments per hour

In [ ]:
avg_by_hour = []

for horas in counts_by_hour: # {9: 45, 13: 85,...}
    avg = comments_by_hour[horas] / counts_by_hour[horas]
    avg_by_hour.append([horas, avg]) #list, and the tuple.
    
avg_by_hour # result in another tupla

With this format its difficult to identify the hours with the highest values:

        [[9, 5.5777777777777775],
         [13, 14.741176470588234],
         [10, 13.440677966101696],
         [14, 13.233644859813085],
         ...
         ]

Let's sorting the list of lists `avg_by_hour` 

### Sorting the `avg_by_hour`.

<a href='#0'> back to index</a>
* * *
## <a id='5'>5. Posts created at a certain time, receive more comments on average </a>

In [ ]:
sorted_avg_by_hour = sorted(avg_by_hour, reverse=True)
sorted_avg_by_hour[0:5]

### Formating `avg_by_hour` time properly.

In [ ]:
date_format = "%H"

#15:00: 38.59 average comments per hour.

#print("Top 5 Hours for Ask Posts Comments:",'\n')

for row in sorted_avg_by_hour:
    horas = row[0]
    hora = str(horas)
    objeto_datetime =  dt.datetime.strptime(hora, date_format)
    hora = objeto_datetime.hour
    
    avg_comment = row[1]

    template = "At {h}:00 the average comments per post is {c:.2f}".format(h = horas, c = avg_comment )
    
    print(template)

In [ ]:
date_format = "%H" 

swap_avg_by_hour = []

for row in sorted_avg_by_hour: #avg_by_hour
    hora = str(row[0])
    objeto_datetime =  dt.datetime.strptime(hora, date_format)
    horas = objeto_datetime.hour 
    
    puntos = row[1]
    
    tupla = (puntos,hora)
    swap_avg_by_hour.append(tupla)

In [ ]:
for index in sorted(swap_avg_by_hour,reverse=True):
    template = "The average comment per post is {avg:.2f} at {hour}:00".format(avg=index[0], hour=index[1])
    print(template)

<a href='#0'> back to index</a>
* * *
## <a id='6'>6. The most relevant authors in `ask_posts`.</a>

Within a community there is always a group of people who have written a larger number of posts. 

Let's see who are those who have written in `Ask post` as those who have done it in `Show post`

In [ ]:
def most_relevant_authors(kind_of_post):
    author = {}
    sort_dict = []
    
    for row in kind_of_post:
        who = row[5]
        
        if who in author:
            author[who] +=1
        else:
            author[who] = 1
            
    for (key,value) in author.items():
        sort_dict.append((value, key))
    
    return sorted(sort_dict, reverse = True)

### Most relevant authors in `Ask_posts` by number of times posted on the forum.

In [ ]:
most_relevant_authors(ask_posts)

### Most relevant authors in `Show_posts`

In [ ]:
most_relevant_authors(show_posts)

In order not to extend too much we will stay with the first 3 in `Ask_post` since the difference between the second and the third is not only 7 points.

    - hoodoof
    - tmaly
    - whoishiring


And we will do the same in `Show_post`however here the difference is not so much between one score and another, just 3 of difference.

    - vipul4vb
    - soheil
    - max0563

Ordered in such a way that the first one is the one who has made the most posts.

<a href='#0'> back to index</a>
* * *
##  <a id='7'>7. The most significant activity by user in `Ask_posts`</a>

In [ ]:
# ['id', 'title', 'url', 'num_points', 'num_comments', 'author', 'created_at']

def all_activity_post(name, posts):
    datum_format = "%m/%d/%Y %H:%M"
    time_format = '%Y/%m/%d'
    time = 0
    num_points = 0
    post_published = {}
    scoring = 0

    print("Score Date           Post ")
    for row in posts:
        if row[5] == name:           # ['author']
            time +=1
            scoring += int(row[3])
            num_points = int(row[3]) # ['num_points']
            tittle = row[1]
            
            ask_date = row[6]       # 'created_at'
            
            ask_date = dt.datetime.strptime(ask_date, datum_format) # date_format = "%m/%d/%Y %H:%M"
            date_object = ask_date.strftime(time_format)           # time_format = '%Y/%m/%d'
            
            post_published = "{points} -- " "{date} " " {tittle},".format(tittle = row[1],
                                                                          date = date_object, points = row[3] )
            
            print(post_published)
    
    total_number_post = "Total number of post = {n_post}".format(n_post = time)
    avg_score = "Score average = {score:.2f}".format(score=scoring/time)
    print("\n")
    print(total_number_post)
    print(avg_score)

- **Activity** in the forum per user.

- **Score** each post you created.

- **Date**

- Title of the post.

- Total **number of posts** by user.

In [ ]:
all_activity_post('hoodoof', ask_posts)

In [ ]:
all_activity_post('tmaly', ask_posts)

In [ ]:
all_activity_post('tixocloud', ask_posts)

<a href='#0'> back to index</a>
* * *
## <a id='8'>8. The most significant activity by user in `Show_posts`</a>

In [ ]:
all_activity_post('vipul4vb', show_posts)

In [ ]:
all_activity_post('soheil', show_posts)

In [ ]:
all_activity_post('brakmic', show_posts) 

<a href='#0'> back to index</a>
* * *
## <a id='9'>9. Who has been the most voted, when, and title of the post</a>

We note that not all publications have the same value for the community and we see this reflected in the votes they receive, so it would be interesting to know which of the publications that our users made had more votes and therefore which were the most important issues for the community. 

In [ ]:
# ['id', 'title', 'url', 'num_points', 'num_comments', 'author', 'created_at']

def author_most_valuable_post(name, posts):
    datum_format = "%m/%d/%Y %H:%M"
    num_points = 0
    post_published = {}

    print("Score Date           Post ")
    for row in posts:
        if row[5] == name: #author
            if num_points < int(row[3]): #num points
                num_points = int(row[3])

                tittle = row[1]
                
                time_format = '%Y/%m/%d'
                ask_date = row[6]
                ask_date = dt.datetime.strptime(ask_date, datum_format)
                date_object = ask_date.strftime(time_format)
                
                post_published.setdefault(num_points,[]).append(num_points)
                post_published.setdefault(num_points,[]).append(date_object)
                post_published.setdefault(num_points,[]).append(tittle)
                
    for i in reversed(post_published):
        print(post_published[i])

### `Ask_post`

In [ ]:
author_most_valuable_post('hoodoof', ask_posts)

In [ ]:
author_most_valuable_post('tmaly', ask_posts)

In [ ]:
author_most_valuable_post('tixocloud', ask_posts)

### `Show_post`

In [ ]:
author_most_valuable_post('vipul4vb', show_posts)

In [ ]:
author_most_valuable_post('soheil', show_posts)

In [ ]:
brakmic = author_most_valuable_post('brakmic', show_posts)

<a href='#0'> back to index</a>
* * *
## <a id='10'>10. Summary</a>

A first step that we can take to know the community is to separate the posts that are of question type `ask_post` and those that are of an informative nature for the community `show_show`

    avg number of "Ask_posts" comments: 14.04
    max number of "Ask_posts" comment: 947

    avg number of "Ask_posts" comments: 10.32
    max number of "Ask_posts" comment: 306


As the importance of the posts on the forum is determined by the maximum score, seeing what are the maximum scores that both posts, we can get an idea of what are the issues that concern this community.

 |`Ask_posts`||
 |:---|:---|
 |**-- Score --**|    **-- Tittle --**|
     |947 |    Ask HN: Who is hiring? (August 2016)|
     |910 |    Ask HN: Who is hiring? (September 2016)|
     |266 |    Ask HN: What are the must-read books about economics/finance?|
     |250 |    Ask HN: Who wants to be hired? (June 2016)|
     |234 |    Ask HN: What are you currently building?|
     |182 |    Ask HN: What is your go-to example for a good REST API?|
     |37  |   Ask HN: Things you created in 2015?|
     |33  |   Ask HN: teaching basic coding and web design offline, solely via iOS devices?|
     |29  |   Ask HN: Am I the only one outraged by Twitter shutting down share counts?|
     |6   |  Ask HN: How to improve my personal website?| 

|`Show_posts`||     
|:---|:---|
|**-- Score --**|    **-- Tittle --**| 
    |306|           Show HN: BitKeeper  Enterprise-ready version control, now open-source|
    |168|           Show HN: Nodal. Next-Generation Node.js Server and Framework|
    |134|           Show HN: Download any song without knowing its name|
    |102|           Show HN: Something pointless I made|
    |22 |          Show HN: Wio Link  ESP8266 Based Web of Things Hardware Development Platform|

The activity in `Ask posts` post is greater than in `Show post`, therefore now we will focus on `Ask posts`. 

Determining the best time to write a question on the forum it is important if you want to be read by the largest number of users, therefore having a ratio of average values per hour gives us an idea of the activity of the forum.

|Average comments per post|Hour|
|:---:|:---|
| **38.59**| 15:00|
| **23.81**| 2:00|
| **21.52**  | 20:00|
| 16.80  | 16:00|
| 16.01  | 21:00|
| 14.74  | 13:00|
| 13.44  | 10:00|
| 13.23  | 14:00|
| 13.20  | 18:00|
| 11.46  | 17:00|
| 11.38  | 1:00|
| 11.05  | 11:00|
| 10.80  | 19:00|
| 10.25  | 8:00|
| 10.09  | 5:00|
| 9.41   |12:00|
| 9.02   |6:00|
| 8.13   |0:00|
| 7.99   |23:00|
| 7.85   |7:00|
| 7.80   |3:00|
| 7.17   |4:00|
| 6.75   |22:00|
| 5.58   |9:00|

By choosing one of the first three hours we can affirm that writing at that time will allow us to have more possibilities of being read.

Ranking of Authors by type of post, number of times they have written post, title and rating

|-- post --|-- ranking-- | -- authors --| -- # of post -- | -- Title of most valuable post --- |-- Score -- | -- Date -- |
|:---|:---|: -- |:--- |:-- |:--: |:--:|
|Ask post|1|hoodoof  |16  |Ask HN: Is there an up-to-date global index of conferences?'|11|2016/06/03|
|Ask post|2|tmaly    |14  |'12, '2016/03/21', 'Ask HN: How do you find unused CSS?'|12|2016/06/03|
|Ask post|3|tixocloud|6   |'Ask HN: Would you use an app to keep track of all your relationships?'|8|'2016/01/07'|
|---|---|---|---|---|---|---|
|Show post|1|vipul4vb  |4  |'Show HN: Quantitative user research reveals useful UX observation on LinkedIn'|2|'2016/04/04'|
|Show post|2|soheil  |4  |'Ask HN: How do you find unused CSS?'|7|2016/03/06|
|Show post|3|brakmic |2  |'Show HN: PouchDB Bindings for PureScript'|21|'2016/02/27'|

In every group there are people who are important, people within the community. 

Identifying them allows us to know a little more about this place and those who feel comfortable in this place.

However, it would be easy to think that the person who writes most often is the most relevant person in the group, the quality of what is written is even more important and it can be seen in point 6 where in third place user has an average score that exceeds who is in 2nd position.


|**#3tixocloud**                 |**#2tmaly**|
|:---|:----|
|Total number of post = 6  |Total number of post = 14|
|Score average = 3.67      |Score average = 3.14|


This reveals that a person who having written less than half exceeds, the one who has written twice. 

Their **comments are much better rated** by the community and that each of them brings much more value than those people who write more frequently.

Having technological thinking is rewarded and adding value is one of the keys in this community

Finally and as a curiosity, knowing what was the maximum score, by user when it happend and what was the title of the post, is a summarized way to make an *scrennshoot* of the community. 

As a final example with the highest number of interventions is `Ask`, the maximum score is given in the `Show` post with **21** points, showing the importance of quality what mostly predominates in this community


therefore to be a member that adds value in the community you should write in the time slots that have been described to be able to reach more people and get points.

it doesn't matter which of the two posts you write in, what matters is the quality of the posts.

and quality is usually determined by a combination of a technical theme that adds value to the larger group